### PR TITLE
Cleanup coreir backend prefix

### PR DESCRIPTION
**Pull request by rsetaluri**
Bunch of cleaning up. Branch `cleanup-coreir-backend` has some continued work that might require more thought, so I wanted to get in the beginning of this sooner than later, and keep working on that.

### CODE REVIEW

File: magma/frontend/coreir_.py [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
-    return CoreIRContextSingleton().get_instance()
+    return coreir_context()
+
 
 def ResetCoreIR():
-    CoreIRContextSingleton().reset_instance()
+    reset_coreir_context()
     backend = GetCoreIRBackend()
+    # NOTE(rsetaluri): Is this next line necessary? Don't think it is.
     backend.reset()
```
Comment (leonardt) [inline magma/frontend/coreir_.py:23]: This calls the backend._init function which resetts all the dictionaries/sets which I think is necessary.  For example, the constant cache holds references to modules for constants.  If we reset the context, we need to clear the backend context.

File: magma/ref.py [hunk 2]
```diff
@@ -207,8 +208,17 @@ def qualifiedname(self, sep="."):
...
         return str(self.view.port.name)
+
+
+def _get_root(ref: Ref) -> typing.Optional[Ref]:
+    parent = ref.parent()
+    if parent is None:
+        return None
+    if parent is ref:
+        return ref
+    return _get_root(parent)
```
Comment (leonardt) [inline magma/ref.py:224]: Hmm, since this is called with self, is there a reason it's not just a private instance method?

**Review by leonardt — APPROVED** [review]

File: magma/frontend/coreir_.py (hunk 1 above)
Comment (rsetaluri) [reply magma/frontend/coreir_.py:24]: Updated comment.

File: magma/ref.py (hunk 2 above)
Comment (rsetaluri) [reply magma/ref.py]: Good point; refactored

**Pull request merged**